lib/rules/no-template-curly-in-string.js

```diff
@@ -22,23 +22,43 @@ module.exports = {
         schema: [],
 
         messages: {
-            unexpectedTemplateExpression: "Unexpected template string expression."
-        }
+            unexpectedTemplateExpression: "Unexpected template string expression.",
+            convertToTemplate: "Convert to ES6 `template`."
+        },
+        hasSuggestions: true
     },
 
     create(context) {
         const regex = /\$\{[^}]+\}/u;
+        const sourceCode = context.getSourceCode();
 
         return {
             Literal(node) {
                 if (typeof node.value === "string" && regex.test(node.value)) {
                     context.report({
                         node,
-                        messageId: "unexpectedTemplateExpression"
+                        messageId: "unexpectedTemplateExpression",
+                        suggest: [
+                            {
+                                messageId: "convertToTemplate",
+                                fix: fixer => {
+                                    const text = sourceCode.getText(node);
+                                    const oldQuote = text[0];
+                                    const textNoQuotes = text.slice(1, -1);
+                                    const escapeBackticks = textNoQuotes.replace(/`/gui, "\\`");
```

Comment:
I'm not sure what exactly CodeQL expects here, but this doesn't check if the backtick was already escaped.

```js
const textNoQuotes = "\\`";
const escapeBackticks = textNoQuotes.replace(/`/gui, "\\`");

console.log(textNoQuotes); // \`
console.log(escapeBackticks); // \\`
```

For example, when the suggestion fixes this string:

```js
/* eslint no-template-curly-in-string: "error" */

var foo = "\`${bar}";
```

it will produce a template literal with escaped backslash, but the backtick isn't escaped:

```js
/* eslint no-template-curly-in-string: "error" */

var foo = `\\`${bar}`; // Parsing error, unexpected token $

```